common-pg/src/main/scala/com/socrata/pg/analyzer2/SoQLFunctionSqlizer.scala

```diff
@@ -0,0 +1,357 @@
+package com.socrata.pg.analyzer2
+
+import com.socrata.prettyprint.prelude._
+import com.socrata.soql.analyzer2._
+import com.socrata.soql.types._
+import com.socrata.soql.functions.SoQLFunctions._
+import com.socrata.soql.functions.{Function, SoQLTypeInfo}
+
+class SoQLFunctionSqlizer[MT <: MetaTypes with ({ type ColumnType = SoQLType; type ColumnValue = SoQLValue })] extends FuncallSqlizer[MT] {
+  import SoQLTypeInfo.hasType
+
+  def wrap(e: Expr, exprSql: ExprSql, wrapper: String, additionalWrapperArgs: Doc*) =
+    ExprSql((exprSql.compressed.sql +: additionalWrapperArgs).funcall(Doc(wrapper)), e)
+
+  def sqlizeNormalOrdinaryWithWrapper(name: String, wrapper: String) = ofs { (f, args, ctx) =>
+    val exprSql = sqlizeNormalOrdinaryFuncall(name)(f, args, ctx)
+    wrap(f, exprSql, wrapper)
+  }
+
+  def sqlizeNormalAggregateWithWrapper(name: String, wrapper: String) = afs { (f, args, filter, ctx) =>
+    val exprSql = sqlizeNormalAggregateFuncall(name)(f, args, filter, ctx)
+    wrap(f, exprSql, wrapper)
+  }
+
+  def sqlizeMultiBuffered(name: String) = ofs { (f, args, ctx) =>
+    val exprSql = sqlizeNormalOrdinaryFuncall(name)(f, args, ctx)
+    wrap(f, wrap(f, exprSql, "st_buffer", d"0.0"), "st_multi")
+  }
+
+  def sqlizeGeomCast(sqlFunctionName: String) = ofs { (f, args, ctx) =>
+    // Just like a normal ordinary function call, but with an
+    // additional synthetic parameter for SRID
+    //
+    // TODO: these casts return NULL on
+    // valid-but-not-actually-this-type WKT.  Might be better to raise
+    // an error then?  Would have to create a temporary for the
+    // argument if it's nontrivial.  Won't happen if the cast is being
+    // introduced implicitly.
+    assert(f.function.minArity == 1 && !f.function.isVariadic)
+    assert(f.function.allParameters == args.map(_.typ))
+
+    val sql = (args.map(_.compressed.sql) :+ d"4326").funcall(Doc(sqlFunctionName))
```

Comment:
Inclined to say `4325` should be something named, rather than a magic number in the middle of nowhere.